src/controllers/user.controller.js

```
@@ -0,0 +1,177 @@
+import { v4 as uuidv4 } from 'uuid';
+import { emailService } from './email.service.js';
+
+const getAll = async (req, res) => {
+  try {
+    const users = await userService.getAllActive();
+    const normalized = users.map(userService.normalize);
+
+    return res.json(normalized);
+  } catch (error) {
+    return res.status(401).json({
+      message: error.message,
+    });
+  }
+};
+
+const updateNameOrPassword = async (req, res) => {
+  const { userId } = req.params;
+  const { userName, pwd, newPwd, confrm } = req.body;
+
+  const updateData = {};
+  let isPasswordValid = false;
+
+  if (!userId) {
+    return res.status(400).send({
+      message: 'not found require params',
+    });
+  }
+
+  const userData = await userService.getByUserId(userId);
+
+  if (!userData) {
+    return res.status(404).send({
+      message: 'user not found',
+    });
+  }
+
+  const userNameNormalized = (userName || '').trim();
+
+  if (userNameNormalized) {
+    const NameValidErrorMessage = validate.name(userNameNormalized);
+
+    if (!NameValidErrorMessage) {
+      updateData.userName = userNameNormalized;
+    } else {
+      return res.status(400).send({
+        message: NameValidErrorMessage,
+      });
+    }
+  }
+
+  if (pwd) {
+    isPasswordValid = await bcrypt.compare(pwd, userData.password);
+
+    if (!isPasswordValid) {
+      return res.status(400).send({
+        message: 'need current password',
+      });
+    }
+  }
+
+  if (pwd && newPwd && confrm) {
+    const passwordVaildError = validate.password(newPwd);
+    const isComare = newPwd === confrm;
+
+    if (passwordVaildError || !isComare) {
+      return res.status(400).send({
+        message: 'Invalid new password or confrm',
+      });
+    }
+
+    if (isPasswordValid && !passwordVaildError && isComare) {
+      const saltRounds = 10;
+      const hashedPassword = await bcrypt.hash(newPwd, saltRounds);
+
+      updateData.password = hashedPassword;
+    }
+  }
```

According to the requirements, changing a password requires an old password, a new password, and a confirmation. This logic block only runs if all three fields (`pwd`, `newPwd`, `confrm`) are provided. If a user sends `newPwd` and `confrm` but omits `pwd`, the password change is silently ignored instead of returning an error. Consider adding a check to ensure that if a password change is attempted (i.e., `newPwd` is present), the old password (`pwd`) is also required.